contracts/utils/Memory.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Helper library for deallocating memory reserved by abi.encode or low level calls.
+ */
+library Memory {
+    type FreePtr is bytes32;
+
+    function save() internal pure returns (FreePtr ptr) {
```

Comment:
Similarly, I think this should be very explicit, otherwise reading `Memory.save()` doesn't tell much.

```suggestion
    function saveFreePointer() internal pure returns (Pointer ptr) {
```